src/quest/historian.py

```diff
@@ -186,12 +186,12 @@ def get_function_name(func):
 
 
 class Historian:
-    def __init__(self, workflow_id: str, workflow: Callable, history: History, master_serializer: MasterSerializer):
+    def __init__(self, workflow_id: str, workflow: Callable, history: History, serializer: StepSerializer = None):
         # TODO - change nomenclature (away from workflow)? Maybe just use workflow.__name__?
         self.workflow_id = workflow_id
         self.workflow = workflow
         self._configurations: list[tuple[Callable, list, dict]] = []
-        self._master_serializer = master_serializer
+        self._serializer: StepSerializer = serializer or NoopSerializer()
```

Comment:
I think having the default logic in the `create...` functions makes the most sense, and the Historian simply assigns what it was given.